src/components/diagrams/hooks/use-diagram-model.ts

```diff
@@ -0,0 +1,326 @@
+                    .some((d) => d.substationId === diagramParams.substationId);
+            } else if (diagramParams.type === DiagramType.NETWORK_AREA_DIAGRAM) {
+                return Object.values(diagrams)
+                    .filter((diagram) => diagram.type === DiagramType.NETWORK_AREA_DIAGRAM)
+                    .some((d) => diagramParams.voltageLevelIds.every((vlId) => d.voltageLevelIds.includes(vlId)));
+            } else if (diagramParams.type === DiagramType.NAD_FROM_CONFIG) {
+                return Object.values(diagrams)
+                    .filter((diagram) => diagram.type === DiagramType.NAD_FROM_CONFIG)
+                    .some((d) => d.nadFromConfigUuid === diagramParams.nadFromConfigUuid);
+            }
+            return false;
+        },
+        [diagrams]
+    );
+
+    const createDiagram = useCallback(
+        (diagramParams: DiagramParams) => {
+            if (filterDiagramParams([diagramParams]).length === 0) {
+                // this hook instance don't manage this type of diagram
+                return;
+            }
+            if (diagramAlreadyExists(diagramParams)) {
+                // blink the diagram
+                return;
+            }
+            const diagram = createPendingDiagram(diagramParams);
+            fetchDiagramSvg(diagram);
+        },
+        [createPendingDiagram, diagramAlreadyExists, fetchDiagramSvg, filterDiagramParams]
+    );
+
+    const removeDiagram = useCallback((id: UUID) => {
+        setDiagrams((oldDiagrams) => {
+            if (oldDiagrams[id]) {
+                const newDiagrams = { ...oldDiagrams };
+                delete newDiagrams[id];
+                return newDiagrams;
+            }
+            return oldDiagrams;
+        });
+    }, []);
+
+    useDiagramEventListener({ createDiagram, removeDiagram });
+    useDiagramNotificationsListener({ updateAllDiagrams });
+    useDiagramSessionStorage({ diagrams, onLoadFromSessionStorage: createDiagram });
+
+    useEffect(() => {
+        if (currentNode?.id === prevCurrentNodeId.current) {
+            return;
+        }
+        // Logic to update the diagrams when the current node changes
+        prevCurrentNodeId.current = currentNode?.id;
+        updateAllDiagrams();
+    }, [currentNode, updateAllDiagrams]);
+
+    useEffect(() => {
+        if (!isStatusBuilt(currentNode?.data?.globalBuildStatus) || isStatusBuilt(prevCurrentNodeStatus.current)) {
+            return;
+        }
+        // Logic to update the diagrams when the current node satus changes
+        prevCurrentNodeStatus.current = currentNode?.data?.globalBuildStatus;
+        updateAllDiagrams();
+    }, [currentNode, updateAllDiagrams]);
+
+    useEffect(() => {
+        if (currentRootNetworkUuid === prevcurrentRootNetworkUuid.current) {
+            return;
+        }
+        // Logic to update the diagrams when the current root network changes
+        prevcurrentRootNetworkUuid.current = currentRootNetworkUuid;
```

Comment:
prevCurrentNodeStatus
prevcurrentRootNetworkUuid

Maybe rename these ? Maybe unify the use of capitalization ?

Reply:
true